frontend/src/App.tsx

```diff
@@ -64,7 +62,7 @@ const App = () => {
    * useEffect react hook for performing side effects
    * in function components.
    */
-  useEffect(() => {
+  React.useEffect(() => {
```

Comment:
This whole hook could be split into multiple atomic `useEffect` and it would improve the readability significantly.

Reply:
@HRcc I remember having a lot of work with this hook when I was preparing the project. I also lie about dependencies here. I could use some advice on how to do it properly.

Reply:
I think this is something [react-query](https://react-query.tanstack.com/) could help you with. Since we use it in the company it might be good idea to get familiar with it here